build_web_compilers/lib/src/dev_compiler_bootstrap.dart

```diff
@@ -310,6 +310,12 @@ final _baseUrlScript = '''
 // Attempt to detect --precompiled mode for tests, and set the base url
 // appropriately, otherwise set it to "/".
 var baseUrl = (function() {
+  if(document.getElementsByTagName("base")){
+  if(document.getElementsByTagName("base")[0]){
```

Comment:
nit: only do one query on the document and assign it to a local variable

Reply:
If we don't do it this way then it will cause an error if the element is not found. I don't know how else we can implement this without causing an error.

Reply:
not sure why it would cause an error...

```js
var baseTag = document.getElementsByTagName("base");
if(baseTag && baseTag[0] && baseTag[0].href){
  return baseTag[0].href;
}
```